"    def create_network(self, instance_name, instance, network_info):        """Create the LXD container network on the host        :param instance_name: nova instance name        :param instance: nova instance object        :param network_info: instance network configuration object        :return:network configuration dictionary        """        LOG.debug('create_network called for instance', instance=instance)        try:            network_devices = {}            if not network_info:                return            for vifaddr in network_info:                cfg = self.vif_driver.get_config(instance, vifaddr)                network_devices[str(cfg['bridge'])] = \                    {'nictype': 'bridged',                     'hwaddr': str(cfg['mac_address']),                     'parent': str(cfg['bridge']),                     'type': 'nic'}                return network_devices        except Exception as ex:            with excutils.save_and_reraise_exception():                LOG.error(                    _LE('Fail to configure network for %(instance)s: %(ex)s'),                    {'instance': instance_name, 'ex': ex}, instance=instance)"